src/index.js

```diff
@@ -2,108 +2,120 @@
 // @flow
 import qs from 'qs'
 import merge from 'lodash.merge'
-import ponyfill from 'fetch-ponyfill'
 
-type OptionsRequest = {
+type Method = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE' | 'HEAD'
+
+type AdapterRequest = {
   abort: () => void,
   promise: Promise<*>
 }
 
-type Method = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE'
-type Options = {
+type RequestOptions = {
   method: Method,
-  headers?: ?{ [key: string]: string },
-  onProgress?: (num: number) => mixed,
-  data?: ?{ [key: string]: mixed }
+  headers?: { [string]: string },
+  data?: { [string]: mixed },
+  qs?: { [string]: mixed },
 }
 
-export function ajaxOptions (options: Options): any {
-  let HeadersConstructor = Headers
-  const { headers, data, ...otherOptions } = options
+type RequestResponse = {
+  ok: boolean,
+  json(): mixed
+}
 
-  if (typeof HeadersConstructor === 'undefined') {
-    HeadersConstructor = ponyfill().Headers
-  }
+type RequestError = {
+  errors?: mixed
+}
 
+export function ajaxOptions (options: RequestOptions): any {
+  const { headers, data, ...otherOptions } = options
   const baseHeaders = {}
+
   if (data) {
     baseHeaders['Content-Type'] = 'application/json'
   }
 
-  const headersObject = new HeadersConstructor(
+  const headersObject = new Headers(
     Object.assign(baseHeaders, headers)
   )
 
   return {
     ...otherOptions,
     headers: headersObject,
-    body: data ? JSON.stringify(data) : null
+    body: data ? JSON.stringify(data) : undefined
```

Comment:
just to understand, why this change?

Reply:
For this https://github.com/github/fetch/issues/402. Particularly, Edge was failing with the error `Failed to execute 'fetch()' on 'Window': HEAD or GET Requests cannot have a body`.

Related: https://github.com/agraboso/redux-api-middleware/issues/138